src/util.c

```diff
@@ -923,6 +924,54 @@ sds makePath(char *path, char *filename) {
     return sdscatfmt(sdsempty(), "%s/%s", path, filename);
 }
 
+/* Given the filename, sync the corresponding directory.
+ *
+ * Usually a portable and safe pattern to overwrite existing files would be like:
+ * 1. create a new temp file (on the same file system!)
+ * 2. write data to the temp file
+ * 3. fsync() the temp file
+ * 4. rename the temp file to the appropriate name
+ * 5. fsync() the containing directory */
+int fsyncFileDir(const char *filename) {
+#ifdef _AIX
+    /* AIX is unable to fsync a directory */
+    return 0;
+#endif
+    char temp_filename[PATH_MAX + 1];
+    char *dname;
+    int dir_fd;
+
+    if (strlen(filename) > PATH_MAX) {
+        errno = ENAMETOOLONG;
+        return -1;
+    }
+
+    /* In the glibc implementation dirname may modify their argument. */
+    memcpy(temp_filename, filename, strlen(filename) + 1);
+    dname = dirname(temp_filename);
+
+    dir_fd = open(dname, O_RDONLY);
+    if (dir_fd == -1) {
+        /* Some OSs don't allow us to open directories at all (Windows returns
+         * EACCES), just ignore the error in that case */
+        if (errno == EISDIR || errno == EACCES) {
```

Comment:
We shouldn't ignore EACCESS:
1. Redis is not supported on Windows anyway.
2. It's a valid error if we have `w+x` on the directory: we'll be able to create/rename files but not read.

Reply:
It was just not obvious from the comment if any other system returns EACCES if opening a directory is not allowed. Anyway, case-2 is valid indeed and probably no need to be paranoid about this and no need to ignore EACCESS.